api/proxy/proxy.go

```diff
@@ -405,3 +427,33 @@ func (p *proxy) applyDefaultForNonDAVRequest(val api.DefaultNonDAVRequest, w htt
 		return
 	}
 }
+
+func (p *proxy) validClient(userAgent string) bool {
```

Comment:
Let's simplify stuff here.

The endpoint is not used only by sync clients so we need to be open. The logic can be:

1) Does the user agent matches mirall/x.x.x? If no then we let it through, else 2.
2) Is version >= minimun; if yes go trough, else 403.

Reply:
What's the difference between what you wrote and what I implemented?

Reply:
Change the return code to 403.

What happens when minimum version is 2.3.2 and user-agent gives only 2.3?

Reply:
is rejected because `2.3` is considered `2.3.0`